pkgs/by-name/lm/lmms/package.nix

```diff
@@ -0,0 +1,117 @@
+{
+  SDL2,
+  alsa-lib,
+  carla,
+  cmake,
+  fetchFromGitHub,
+  fftwFloat,
+  fltk,
+  fluidsynth,
+  glibc_multi,
+  lame,
+  lib,
+  libgig,
+  libjack2,
+  libogg,
+  libpulseaudio,
+  libsForQt5,
+  libsamplerate,
+  libsndfile,
+  libsoundio,
+  libvorbis,
+  lilv,
+  lv2,
+  perl540,
+  perl540Packages,
+  pkg-config,
+  portaudio,
+  qt5,
+  sndio,
+  stdenv,
+  substitute,
+  suil,
+  wineWowPackages,
+  withOptionals ? false,
+}:
+
+let
+  winePackage = if lib.isDerivation wineWowPackages then wineWowPackages else wineWowPackages.minimal;
+in
+stdenv.mkDerivation {
+  pname = "lmms";
+  version = "0-unstable-2025-01-28";
+
+  src = fetchFromGitHub {
+    owner = "LMMS";
+    repo = "lmms";
+    rev = "2c674eca3accbc6dab6f5aacbf34f7bd85af767e";
+    sha256 = "sha256-jq3Wog5BHBWIq5iLEBrHWIvYSzmKsCdb63Nu7j9kJRY=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    libsForQt5.qt5.qttools
+    pkg-config
+    qt5.wrapQtAppsHook
+  ];
+
+  buildInputs =
+    [
+      fftwFloat
+      libsForQt5.qt5.qtbase
+      libsForQt5.qt5.qtx11extras
+      libsamplerate
+      libsndfile
+    ]
+    ++ lib.optionals withOptionals [
+      SDL2
+      alsa-lib
+      carla
+      fltk
+      fluidsynth
+      glibc_multi
+      lame
+      libgig
+      libjack2
+      libogg
+      libpulseaudio
+      libsoundio
+      libvorbis
+      lilv
+      lv2
+      perl540
+      perl540Packages.ListMoreUtils
+      perl540Packages.XMLParser
+      portaudio
+      sndio
+      suil
+      winePackage
+    ];
```

Comment:
having the default package be without any audio doesn't make sense tbh.
i'd go for having a sensible default package and then allowing people to create the own version by calling the derivation with different arguments.

Reply:
like if you really wanted to make the most use of this you could have separate flags for jack, alsa and pulseaudio and then a combined withAudio flag that is the true by default.